plugins/terminal.koplugin/main.lua

```diff
@@ -1,58 +1,394 @@
+    UIManager:show(InfoMessage:new {
+        text = _("Shortcuts saved!"),
+        timeout = 3
+    })
+end
+
+function Terminal:manageShortcuts()
+    self.shortcuts_dialog = CenterContainer:new {
+        dimen = Screen:getSize(),
+    }
+    self.shortcuts_menu = Menu:new {
+        show_parent = self.ui,
+        width = Screen:getWidth(),
+        height = Screen:getHeight(),
+        covers_fullscreen = true, -- hint for UIManager:_repaint()
+        is_borderless = true,
+        is_popout = false,
+        perpage = self.items_per_page,
+        onMenuHold = self.onMenuHoldShortcuts,
+        _manager = self,
+    }
+    table.insert(self.shortcuts_dialog, self.shortcuts_menu)
+    self.shortcuts_menu.close_callback = function()
+        UIManager:close(self.shortcuts_dialog)
+    end
+
+    -- sort the shortcuts:
+    if #self.shortcuts > 0 then
+        table.sort(self.shortcuts, function(v1, v2)
+            return v1.text < v2.text
+        end)
+    end
+    self:updateItemTable()
+end
+
+function Terminal:updateItemTable()
+    local item_table = {}
+    if #self.shortcuts > 0 then
+        local actions_count = 3 -- separator + actions
+        for nr, f in ipairs(self.shortcuts) do
+            local item = {
+                nr = nr,
+                text = f.text,
+                commands = f.commands,
+                editable = true,
+                deletable = true,
+                callback = function()
+                    -- so we know which middle button to display in the results:
+                    self.source = "shortcut"
+                    -- execute immediately, skip terminal dialog:
+                    self.command = f.commands
+                    Trapper:wrap(function()
+                        self:execute()
+                    end)
+                end
+            }
+            table.insert(item_table, item)
+            -- add page actions at end of each page with shortcuts:
+            local factor = self.items_per_page - actions_count
+            if nr % factor == 0 or nr == #self.shortcuts then
+                -- insert "separator":
+                table.insert(item_table, {
+                    text = " ",
+                    deletable = false,
+                    editable = false,
+                    callback = function()
+                        self:manageShortcuts()
+                    end,
+                })
+                -- actions:
+                self:insertPageActions(item_table)
+            end
+        end
+        -- no shortcuts defined yet:
+    else
+        self:insertPageActions(item_table)
+    end
+    local title = #self.shortcuts == 1 and _("Shortcut") or _("Shortcuts")
+    self.shortcuts_menu:switchItemTable(tostring(#self.shortcuts) .. " " .. title, item_table)
+    UIManager:show(self.shortcuts_dialog)
+end
+
+function Terminal:insertPageActions(item_table)
+    table.insert(item_table, {
+        text = "   " .. string.lower(_("to terminal")) .. "...",
```

Comment:
Not sure you need to lower() them (I guess I wouldn't mind "To terminal", even without the indentation) - and you don't know how that will work with translations.
<kbd>![image](https://user-images.githubusercontent.com/24273478/91730233-fa518880-eba5-11ea-8123-a81b72fd50de.png)</kbd>.
And make the "..." a proper ellipsis (single char), and keep it part of the translated string (we don't know how some other languages might prefer that)

Reply:
> Not sure you need to lower() them (I guess I wouldn't mind "To terminal", even without the indentation) - and you don't know how that will work with translations.
> ![image](https://user-images.githubusercontent.com/24273478/91730233-fa518880-eba5-11ea-8123-a81b72fd50de.png).
> And make the "..." a proper ellipsis (single char), and keep it part of the translated string (we don't know how some other languages might prefer that)

I like the lower case and indentation better, for clearer separation from regular shortcuts (as you might have guessed).

Reply:
What's the string.lower about when you're already giving it lowercase characters? It'll just mess up a poor little language like German. ;-)